From 575bf807ecbe28617efe6f8228b97bc4b6cbf5d0 Mon Sep 17 00:00:00 2001
From: shorinkiwata <fcl709@outlook.com>
Date: Sun, 11 Oct 2026 20:28:17 +0900
Subject: [PATCH 37/42] Import silence-starcitizen-unsupported-os from
 GE-Proton

Imported from GE-Proton11-7 patches/game-patches/silence-starcitizen-unsupported-os.patch.
---
 dlls/user32/msgbox.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/dlls/user32/msgbox.c b/dlls/user32/msgbox.c
index 009f3858c60..37956f146d2 100644
--- a/dlls/user32/msgbox.c
+++ b/dlls/user32/msgbox.c
@@ -400,6 +400,7 @@ INT WINAPI MessageBoxW( HWND hwnd, LPCWSTR text, LPCWSTR title, UINT type )
     return MessageBoxExW(hwnd, text, title, type, LANG_NEUTRAL);
 }
 
+const char *sc_unsupported = "You're trying to run the game on an unsupported Windows OS. This will likely result in undefined behavior.";
 
 /**************************************************************************
  *		MessageBoxExA (USER32.@)
@@ -409,6 +410,9 @@ INT WINAPI MessageBoxExA( HWND hWnd, LPCSTR text, LPCSTR title,
 {
     MSGBOXPARAMSA msgbox;
 
+    if (title && text && strcmp(title, "Star Citizen") == 0 && strcmp(text, sc_unsupported) == 0)
+            return 0;
+
     msgbox.cbSize = sizeof(msgbox);
     msgbox.hwndOwner = hWnd;
     msgbox.hInstance = 0;
-- 
2.56.0

